### PR TITLE
useReduce implemented in gamecontext

### PR DESCRIPTION
**Pull request by MatiasGranHenriksen**

### CODE REVIEW

File: web/app/routes/start/index.tsx [hunk 1]
```diff
@@ -15,21 +13,15 @@ export default function StartQuizIndexRoute() {
 
     const quizes: IQuiz[] = useLoaderData()
     const navigate = useNavigate()
-    const { setPinCode } = useQuiz()
-
-    const { setGameProps } = useGameContext()
 
 
     const startQuiz = async (quizId: number | undefined) => {
         const res = await fetch(`http://0.0.0.0:8080/game/${quizId}/gamestart`)
```
Comment (EmelijonJ) [inline web/app/routes/start/index.tsx:19]:
```suggestion
        const res = await fetch(`https://kviss-api.dev.nav.no/game/${quizId}/gamestart`)
```
Maybe change

**Review by EmelijonJ — APPROVED** [review]
I trust the front end boys

File: web/app/context/game/game.ts [hunk 2]
```diff
@@ -5,11 +5,13 @@ export interface Game {
...
 export enum ActionTypes {
     RESET = 'RESET',
+    SET_USERNAME = 'SET_USERNAME',
+    SET_PINCODE = 'SET_PINCODE'
 }
 
 export interface IGameAction {
     type: ActionTypes
-    payload?: Game
+    payload?: any
```
Comment (Henrhh) [inline web/app/context/game/game.ts:14]: Why type any?

**Review by Henrhh — CHANGES_REQUESTED** [review]
Fix type any, make an interface if needed

File: web/app/context/game/game.ts (hunk 2 above)
Comment (MatiasGranHenriksen) [reply web/app/context/game/game.ts]: Fixed, payload set to String temporary

**Review by Henrhh — APPROVED** [review]
sure why not

**Pull request merged**